src/main/scala/beam/router/model/EmbodiedBeamTrip.scala

```diff
@@ -42,6 +42,7 @@ case class EmbodiedBeamTrip(legs: IndexedSeq[EmbodiedBeamLeg], router: Option[St
   lazy val replanningPenalty: Double = legs.map(_.replanningPenalty).sum
 
   val totalTravelTimeInSecs: Int = legs.lastOption.map(_.beamLeg.endTime - legs.head.beamLeg.startTime).getOrElse(0)
+  def totalDistanceInM: Double = legs.view.map(_.beamLeg.travelPath.distanceInM).foldLeft(0.0)(_ + _)
```

Comment:
Can this be a `val`? It seems that legs cannot change?

Reply:
Yes, but in this case it would be calculated every time when an instance is created. And that method is not used every time.

Reply:
How about `lazy val` then?

Reply:
That might work. But `lazy val`s are not so cheap. They are generally for correctness not for optimization as I understood.